lib/checkleakautovar.cpp

```diff
@@ -343,8 +350,8 @@ void CheckLeakAutoVar::checkScope(const Token * const startToken,
             ftok = ftok->tokAt(2);
 
         // assignment..
-        if (Token::Match(varTok, "%var% =")) {
-            const Token* const tokAssignOp = varTok->next();
+        if (Token::Match(varTok, "%var% =") || (varTok->varId() && !Token::simpleMatch(varTok->astParent(), "[") && varTok->astTop()->str() == "=")) {
```

Comment:
I don't think we should use astTop here..

imagine:

    x = f(y,z);

if `varTok` is `z` then we will assume that `z` is assigned here as far as I see.

Reply:
I think I have fixed that issue.
Do you expect problems in other checks when `simplifyMulAndParens()` is no longer applied?

Reply:
I don't spontanously expect further problems. Let's get rid of it :+1: